### DISCUSSION TITLE
Disabling some features in `styles.yml` instead of using commands

### DISCUSSION THREAD

**Discussion by jodros** [Omikhleia/resilient.sile#101] [Q&A] [answered]
Sorry if this is a too silly question, maybe I'll still have many of them while I'm getting used to resilient....

I want to disable headers, and I know about `\noheaders`, but since headers are also personalized in the styles files, and having headers or not is a matter of styling, I'd like to know whether there is any way to disable them through the `-styles.yml` file. I've tried things like 

```yml 
header-base:
    origin: false
```

and similar...

I just wish to do as much as I can without using SILE commands.

**Comment by Omikhleia** [comment, marked as answer]
> having headers or not is a matter of styling

Well, not really (for now):
Having headers or not is a (global) book-type decision; styling only says how to present such things (when present), i.e. font, etc.

In your master document, the `sile` section should looks like the following (plus or minus the optional bits)

```yaml
sile:
  options:
    papersize: 6in x 9in
    layout: ateliers demiluxe
    headers: novel
    resolution: 300
``` 

The `headers` setting (ultimately, a class option) can be tuned as follows, in principle (see also the User Guide, part VI, chap. 1):
 - `none`: No running headers.
 - `technical` (default): Chapter and section titles go in the even and odd running headers, respectively.
 - `novel`: Chapter titles go in the odd running headers, and the title of the book, if set, goes in the even running headers.

So what you want is **none** here.
This being said, at some point I broke something during a refactor ("novel" is impacted, #99), and I haven't investigated yet if "none" still works as expected or not.

**Reply by Omikhleia** [reply]
By the way, there's no such things as too silly questions! But there's a _User Guide_, which needs some love, and still shows too much of the SIL legacy, with a not-so-appropriate organization of the topics (see notably #73). So it's good to see where people err, 'cause documentation writing is hard when you know your own stuff all too well and no longer question yourself :cat:

**Reply by jodros** [reply]
Thank you! I tried to change the headers option as well, but with headers: false instead of none 🤦.

**Reply by jodros** [reply]
Of course I'm always looking for new knowledge in the User Guide, but sometimes what I wanna find goes unnoticed, just now looking on it again I realized this `headers: none` is on the class options, this time it was just my lack of attention...

**Comment by jodros** [comment]
Now I've again other too similar question to create a new thread:

I know one can disable figure counting with `{.unnumbered}`, but what if one wants to disable the captions numbering of every figure?

![изображение](https://github.com/user-attachments/assets/d0fbd30e-7d7c-4fdc-8e62-dc8a015a73ec)

After set the numbering of `figure-caption-base-number`, `figure-caption-main-number` and `figure-caption-ref-number` to false, and I still get the at least a number:

![изображение](https://github.com/user-attachments/assets/a3b2dadb-5854-40bb-9ff9-2525b7865256)

BTW it printed *Figure* even though the document language is set to `pt` and the caption to `ru`, doesn't it  support i18n?

**Reply by Omikhleia** [reply]
> BTW it printed Figure even though the document language (...)

"Figure" is in your style file, from the default generated value. But you can change it :) - Some day we'll also have #89 under the hood, perhaps, to pick pre-customized language dependent styles.

>  I still get the at least a number

Generic numbering disabling was briefly discussed in #24 as you might have noticed (but I'm not sure the proposed way there is the best approach, it has a dirty CSS smell). Until a consensus is reached, explicit `.unnumbered` is the only way.

**Reply by Omikhleia** [reply]
> Until a consensus is reached, explicit `.unnumbered` is the only way.

And I am wrong here, we actually settled on a different solution if the User Guide is correct:

> The `numberstyle` specification defines which number style should be used for the section number, depending on the context where it appears (main text flow, running headers and cross-references). Note that if a given context is absent, the section number will not be displayed.

So one could remove the `main` entry from the `numberstyle` field in the `figure-caption` style, and that should remove the numbering. I tend to forget about those options which I ~~never~~ seldom use myself, so I hope it still works ;)

**Reply by Omikhleia** [reply]
> I tend to forget about those options which I never use myself

Yet how I could forget #21, #22 eludes me lol.

**Reply by jodros** [reply]
It worked, thank you again!
